### PR TITLE
Henry's boundary conditions

### PR DESCRIPTION
**Pull request by gabriele-ferrero**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to FESTIM?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

- [x] Black formatted
- [x] Unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: festim/boundary_conditions/henrys_bc.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        H_0 (float or fem.Constant): the Henrys constant pre-exponential factor (H/m3/Pa)
+        E_H (float or fem.Constant): the Henrys constant activation energy (eV)
+        pressure (float or callable): the pressure at the boundary (Pa)
+
+    Attributes:
+        subdomain (festim.Subdomain): the subdomain where the boundary
+            condition is applied
+        value (float or fem.Constant): the value of the boundary condition
+        species (festim.Species or str): the name of the species
+        H_0 (float or fem.Constant): the Henrys constant pre-exponential factor (H/m3/Pa0.5)
```
Comment (jhdark) [inline festim/boundary_conditions/henrys_bc.py:31]:
```suggestion
        H_0 (float or fem.Constant): the Henrys constant pre-exponential factor (H/m3/Pa)
```

File: test/test_sievertsbc.py [hunk 2]
```diff
@@ -13,6 +13,13 @@ def sieverts_law(T, S_0, E_S, pressure):
     return S * pressure**0.5
 
 
+def test_raise_error():
```
Comment (jhdark) [inline test/test_sievertsbc.py:16]: See above

**Review by jhdark — APPROVED** [review]
Looks great to me, thanks @gabriele-ferrero for the contribution!

**Review by RemDelaporteMathurin — APPROVED** [review]
Thanks @gabriele-ferrero !!

**Pull request merged**